### PR TITLE
[Security Solution Gen AI]Docs for prompts and anonymization fields

### PR DESCRIPTION
**Pull request by YulNaumenko**
Documentation to the public Security API for prompts and anonymization fields added in 8.15

### CODE REVIEW

File: docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+[discrete]
+==== BulkCreateAction object
+
+[width="100%",options="header"]
+|==============================================
+|Name |Type |Description |Required
+
+|`name` |String |Prompt short title. |Yes
+|`id` |String |Prompt ID. |Yes
+|`promptType` |String |Prompt type. Could be 'system' or 'quick' |Yes
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc:36]:
```suggestion
|`promptType` |String |Prompt type.  Valid values are:

* `system`
* `quick`
|Yes
```

File: docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+|`consumer` |String |Solution application to which prompt belongs. |Yes
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc:37]: Can we add some example values here?

File: docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+|`content` |String |Prompt text whixh will be send to LLM. |Yes
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc:38]:
```suggestion
|`content` |String |Prompt text to send to LLM. |Yes
```

File: docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+|`color` |String |Is only for quick prompts to highligh in the Assistant UI. |No
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc:39]:
```suggestion
|`color` |Sets the display color for quick prompts in the UI. |No
```
What are the valid colors?

File: docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+
+[[bulk-create-object-schema]]
+[discrete]
+==== BulkCreateAction object
+
+[width="100%",options="header"]
+|==============================================
+|Name |Type |Description |Required
+
+|`name` |String |Prompt short title. |Yes
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc:34]:
```suggestion
|`name` |String |Short prompt title. |Yes
```

File: docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+[discrete]
+[[bulk-update-object-schema]]
+===== `BulkUpdateAction` schema
+
+[width="100%",options="header"]
+|==============================================
+|Name |Type |Description |Required
+
+|`id` |String |Prompt ID. |Yes
+|`content` |String |Prompt text whixh will be send to LLM. |No
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc:53]:
```suggestion
|`content` |String |Prompt text to send to LLM. |No
```

File: docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+|`color` |String |Is only for quick prompts to highligh in the Assistant UI. |No
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc:54]:
```suggestion
|`color` |String |Sets the display color for quick prompts in the UI. |No
```

File: docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+|`categories` |String[] |Array of prompts categories. |No
+|==============================================
+
+
+[discrete]
+===== Example requests
+
+*Example 1*
+
+The following request create prompt:
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc:64]:
```suggestion
The following request creates a new quick prompt:
```

File: docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+|`consumer` |String |Solution application to which prompt belongs. |Yes
+|`content` |String |Prompt text whixh will be send to LLM. |Yes
+|`color` |String |Is only for quick prompts to highligh in the Assistant UI. |No
+|`categories` |String[] |Array of prompts categories. |No
+|==============================================
+
+
+[discrete]
+[[bulk-update-object-schema]]
+===== `BulkUpdateAction` schema
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc:46]:
```suggestion
==== BulkUpdateAction object
```
Making this the same as the header for BulkCreateAction object. They seem to be comparable objects so figured they should have similar headers. Feel free to ignore if that's inaccurate.

File: docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+- `attributes.summary.total`: Total number of prompts matching the bulk action
+- `attributes.summary.succeeded`: Number of successful outcomes (number of prompts that were created, deleted, or updated)
+- `attributes.summary.failed`: Number of failed outcomes
+- `attributes.summary.skipped`: Number of prompts that were skipped due to various reasons (explained below)
+- `attributes.results.created`: Prompt objects that were created during the action's execution
+- `attributes.results.updated`: Prompt objects that were updated during the action's execution
+- `attributes.results.deleted`: Prompt objects that were deleted during the action's execution
+- `attributes.results.skipped`: Prompt that were skipped during the action's execution
+
+A prompt can only be `skipped` when the bulk action to be performed on it results in nothing being done. For example, if the `update` action is used to add a filed to a prompt that already has that value. Objects returned in `attributes.results.skipped` will only include prompts' `id`, `name`, and `skip_reason`.
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc:104]:
```suggestion
A prompt can only be `skipped` when the bulk action to be performed on it results in nothing being done. For example, if the `update` action is used to add a field to a prompt that already has that value. Objects returned in `attributes.results.skipped` only include a prompt's `id`, `name`, and `skip_reason`.
```

File: docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc [hunk 11]
```diff
@@ -0,0 +1,249 @@
...
+            "failed": 0,
+            "succeeded": 1,
+            "skipped": 0,
+            "total": 1
+        }
+    }
+}
+--------------------------------------------------
+
+*Example 2, Partial failure*
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc:146]:
```suggestion
*Example 2: Partial failure*
```

File: docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc [hunk 12]
```diff
@@ -0,0 +1,249 @@
...
+
+The following request adds tags `tag-1` and `tag-2` to the rules that have the IDs sent in the payload:
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc:148]:
```suggestion
The following request adds tags `tag-1` and `tag-2` to the rules associated with the IDs sent in the payload:
```

File: docs/AI-for-security/api/anonymization-fields-api-find.asciidoc [hunk 13]
```diff
@@ -0,0 +1,269 @@
+[[anonymization-fields-api-find]]
+=== Find AI Assistant anonymization fields
+
+Retrieve a list of Elastic AI anonymization fields.
```
Comment (benironside) [inline docs/AI-for-security/api/anonymization-fields-api-find.asciidoc:4]: I'm not clear on what "anonymization fields" means here. It looks like the response includes fields that are and aren't currently set to be anonymized. "Anonymization fields" sounds like it's just the fields that are set to be anonymized. So what defines an anonymization field?

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 14]
```diff
@@ -0,0 +1,233 @@
...
+
+[[bulk-create-anonymization-fields-schema]]
+[discrete]
+==== BulkCreateAction object
+
+[width="100%",options="header"]
+|==============================================
+|Name |Type |Description |Required
+
+|`field` |String |Field name in the document which will be send to LLM. |Yes
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:34]:
```suggestion
|`field` |String |A field's name. |Yes
```
If I understand correctly, fields may not get sent to the LLM, depending on the `allowed` value. Updated suggestion here accordingly

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 15]
```diff
@@ -0,0 +1,233 @@
...
+|`allowed` |Boolean |Identify if the field should be included to the list of the fields which will be send to LLM. |Yes
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:35]:
```suggestion
|`allowed` |Boolean |Defines whether the field will get sent to the LLM. |Yes
```

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 16]
```diff
@@ -0,0 +1,233 @@
...
+|`anonymized` |Boolean |Indentify if the field should be anonymized. By default is `false` |Yes
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:36]:
```suggestion
|`anonymized` |Boolean |Defines whether the field will be anonymized if sent to the LLM. Defaults to `false` |Yes
```

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 17]
```diff
@@ -0,0 +1,233 @@
...
+
+[discrete]
+[[bulk-update-anonymization-fields-schema]]
+===== `BulkUpdateAction` schema
+
+[width="100%",options="header"]
+|==============================================
+|Name |Type |Description |Required
+
+|`allowed` |Boolean |Identify if the field should be included to the list of the fields which will be send to LLM. |No
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:48]:
```suggestion
|`allowed` |Boolean |Defines whether the field will get sent to the LLM.  |No
```

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 18]
```diff
@@ -0,0 +1,233 @@
...
+|`anonymized` |Boolean |Indentify if the field should be anonymized. By default is `false` |No
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:49]:
```suggestion
|`anonymized` |Boolean |Defines whether the field will be anonymized if sent to the LLM. Defaults to `false`  |No
```

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 19]
```diff
@@ -0,0 +1,233 @@
...
+|==============================================
+
+
+[discrete]
+===== Example requests
+
+*Example 1*
+
+The following request create 2 anonymization fields:
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:58]:
```suggestion
The following request creates 2 anonymization fields:
```

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 20]
```diff
@@ -0,0 +1,233 @@
...
+- `attributes.summary.total`: Total number of anonymization fields matching the bulk action
+- `attributes.summary.succeeded`: Number of successful outcomes (number of anonymization fields that were created, deleted, or updated)
+- `attributes.summary.failed`: Number of failed outcomes
+- `attributes.summary.skipped`: Number of anonymization fields that were skipped due to various reasons (explained below)
+- `attributes.results.created`: Anonymization field objects that were created during the action's execution
+- `attributes.results.updated`: Anonymization field objects that were updated during the action's execution
+- `attributes.results.deleted`: Anonymization field objects that were deleted during the action's execution
+- `attributes.results.skipped`: Anonymization field that were skipped during the action's execution
+
+An anonymization field can only be `skipped` when the bulk action to be performed on it results in nothing being done. For example, if the `update` action is used to add a filed to a prompt that already has that value. Objects returned in `attributes.results.skipped` will only include prompts' `id`, `name`, and `skip_reason`.
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:100]:
```suggestion
An anonymization field can only be `skipped` when the bulk action to be performed on it results in nothing being done. For example, if the `update` action is used to add a field to a prompt that already has that value. Objects returned in `attributes.results.skipped` will only include a prompt's `id`, `name`, and `skip_reason`.
```

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 21]
```diff
@@ -0,0 +1,233 @@
...
+            "failed": 0,
+            "succeeded": 2,
+            "skipped": 0,
+            "total": 2
+        }
+    }
+}
+--------------------------------------------------
+
+*Example 2, Partial failure*
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:143]:
```suggestion
*Example 2: Partial failure*
```

File: docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc (hunk 12 above)
Comment (benironside) [reply docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc:148]: The request doesn't look like it has `tag-1` or `tag-2` in it. I could be missing something but how would it add those tags without mentioning them? And I thought this page was about prompts, not rules?

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 22]
```diff
@@ -0,0 +1,233 @@
...
+
+The following request adds tags `tag-1` and `tag-2` to the rules that have the IDs sent in the payload:
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:145]: Not sure what rules have to do with this, and I'm not sure the example does what this line says (looks like it deletes an anonymization field entry and updates two others). Please confirm, thanks!

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 23]
```diff
@@ -0,0 +1,233 @@
...
+[discrete]
+===== Response code
+
+`500`::
+    Indicates partial bulk action failure.
+
+[discrete]
+===== Response payload
+
+If processing of any rule fails, a partial error outputs the ID and/or name of the affected rule and the corresponding error, as well as successfully processed rules (in the same format as a successful 200 request).
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:177]: What does "rules" mean here? I thought this API was about updating field anonymization settings.

**Review by benironside — COMMENTED** [review]
Great progress! Please let me know if you have questions about any of my comments. Thank you

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc (hunk 23 above)
Comment (YulNaumenko) [reply docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:177]: copied from there

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 24]
```diff
@@ -0,0 +1,233 @@
...
+[discrete]
+===== Request body
+
+A JSON object with the following properties:
+
+[width="100%",options="header"]
+|==============================================
+|Name |Type |Description |Required
+| `create` | <<bulk-create-anonymization-fields-schema, BulkCreateAction[]>> |  Array of anonymization fields objects to create. | No
+| `delete` | String[] |  Array of anonymization fields IDs to which a bulk action will be applied. | No
```
Comment (natasha-moore-elastic) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:20]:
```suggestion
| `delete` | String[] |  Array of IDs of the anonymization fields to delete. | No
```

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 25]
```diff
@@ -0,0 +1,233 @@
...
+| `update` | <<bulk-update-anonymization-fields-schema, BulkUpdateAction[]>> |  Array of anonymization fields objects with the fields to update. | No
```
Comment (natasha-moore-elastic) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:21]:
```suggestion
| `update` | <<bulk-update-anonymization-fields-schema, BulkUpdateAction[]>> |  Array of anonymization field objects with the fields to update. | No
```

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 26]
```diff
@@ -0,0 +1,233 @@
...
+
+|`field` |String |A field's name. |Yes
+|`allowed` |Boolean |Defines whether the field will get sent to the LLM. |Yes
+|`anonymized` |Boolean |Defines whether the field will be anonymized if sent to the LLM. Defaults to `false` |Yes
+|==============================================
+
+
+[discrete]
+[[bulk-update-anonymization-fields-schema]]
+===== `BulkUpdateAction` schema
```
Comment (natasha-moore-elastic) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:42]:
```suggestion
==== BulkUpdateAction object
```

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 27]
```diff
@@ -0,0 +1,233 @@
...
+
+[width="100%",options="header"]
+|==============================================
+|Name |Type |Description |Required
+
+|`allowed` |Boolean |Defines whether the field will get sent to the LLM.  |No
+|`anonymized` |Boolean |Defines whether the field will be anonymized if sent to the LLM. Defaults to `false`  |No
```
Comment (natasha-moore-elastic) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:49]:
```suggestion
|`anonymized` |Boolean |Defines whether the field will be anonymized if sent to the LLM. Defaults to `false`.  |No
```

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 28]
```diff
@@ -0,0 +1,233 @@
...
+[discrete]
+==== BulkCreateAction object
+
+[width="100%",options="header"]
+|==============================================
+|Name |Type |Description |Required
+
+|`field` |String |A field's name. |Yes
+|`allowed` |Boolean |Defines whether the field will get sent to the LLM. |Yes
+|`anonymized` |Boolean |Defines whether the field will be anonymized if sent to the LLM. Defaults to `false` |Yes
```
Comment (natasha-moore-elastic) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:36]:
```suggestion
|`anonymized` |Boolean |Defines whether the field will be anonymized if sent to the LLM. Defaults to `false`. |Yes
```

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 29]
```diff
@@ -0,0 +1,233 @@
...
+         "anonymized":false
+      },
+      {
+         "field":"agent.name",
+         "allowed":true,
+         "anonymized":true
+      }
+   ],
+}
+
```
Comment (natasha-moore-elastic) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:77]:
```suggestion
{
  "create": [
    {
      "field": "agent.type",
      "allowed": true,
      "anonymized": false
    },
    {
      "field": "agent.name",
      "allowed": true,
      "anonymized": true
    }
  ]
}
```

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 30]
```diff
@@ -0,0 +1,233 @@
...
+|`anonymized` |Boolean |Defines whether the field will be anonymized if sent to the LLM. Defaults to `false`  |No
+|==============================================
+
+
+[discrete]
+===== Example requests
+
+*Example 1*
+
+The following request creates 2 anonymization fields:
```
Comment (natasha-moore-elastic) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:58]:
```suggestion
The following request creates two anonymization fields:
```

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 31]
```diff
@@ -0,0 +1,233 @@
...
+JSON object containing the action's outcome:
+
+- `attributes.summary.total`: Total number of anonymization fields matching the bulk action
+- `attributes.summary.succeeded`: Number of successful outcomes (number of anonymization fields that were created, deleted, or updated)
+- `attributes.summary.failed`: Number of failed outcomes
+- `attributes.summary.skipped`: Number of anonymization fields that were skipped due to various reasons (explained below)
+- `attributes.results.created`: Anonymization field objects that were created during the action's execution
+- `attributes.results.updated`: Anonymization field objects that were updated during the action's execution
+- `attributes.results.deleted`: Anonymization field objects that were deleted during the action's execution
+- `attributes.results.skipped`: Anonymization field that were skipped during the action's execution
```
Comment (natasha-moore-elastic) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:98]:
```suggestion
- `attributes.results.skipped`: Anonymization fields that were skipped during the action's execution
```

File: docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc [hunk 32]
```diff
@@ -0,0 +1,241 @@
...
+- `attributes.summary.total`: Total number of prompts matching the bulk action
+- `attributes.summary.succeeded`: Number of successful outcomes (number of prompts that were created, deleted, or updated)
+- `attributes.summary.failed`: Number of failed outcomes
+- `attributes.summary.skipped`: Number of prompts that were skipped due to various reasons (explained below)
+- `attributes.results.created`: Prompt objects that were created during the action's execution
+- `attributes.results.updated`: Prompt objects that were updated during the action's execution
+- `attributes.results.deleted`: Prompt objects that were deleted during the action's execution
+- `attributes.results.skipped`: Prompt that were skipped during the action's execution
+
+A prompt can only be `skipped` when the bulk action to be performed on it results in nothing being done. For example, if the `update` action is used to add a field to a prompt that already has that value. Objects returned in `attributes.results.skipped` only include a prompt's `id`, `name`, and `skip_reason`.
```
Comment (natasha-moore-elastic) [inline docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc:108]:
```suggestion
A prompt can only be `skipped` when the bulk action to be performed on it results in nothing being done. For example, if the `update` action is used to update a field that already has the specified value. Objects returned in `attributes.results.skipped` only include a prompt's `id`, `name`, and `skip_reason`.
```

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 33]
```diff
@@ -0,0 +1,233 @@
...
+
+An anonymization field can only be `skipped` when the bulk action to be performed on it results in nothing being done. For example, if the `update` action is used to add a field to a prompt that already has that value. Objects returned in `attributes.results.skipped` will only include a prompt's `id`, `name`, and `skip_reason`.
```
Comment (natasha-moore-elastic) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:100]: Copy-paste mix-up :) Please check if the suggestion is correct (I don't think anonymization fields have a `name` but not sure what field it should be?)
```suggestion
An anonymization field can only be `skipped` when the bulk action to be performed on it results in nothing being done. For example, if the `update` action is used to update a field that already has the specified value. Objects returned in `attributes.results.skipped` will only include an anonymization field's `id`, `name`, and `skip_reason`.
```

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 34]
```diff
@@ -0,0 +1,233 @@
...
+            "skipped": []
+        },
+        "summary": {
+            "failed": 0,
+            "succeeded": 2,
+            "skipped": 0,
+            "total": 2
+        }
+    }
+}
```
Comment (natasha-moore-elastic) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:140]:
```suggestion
{
  "success": true,
  "anonymization_fields_count": 2,
  "attributes": {
    "results": {
      "updated": [],
      "created": [
        {
          "timestamp": "2024-08-13T20:54:23.125Z",
          "createdAt": "2024-08-13T20:54:23.125Z",
          "field": "agent.type",
          "allowed": true,
          "anonymized": false,
          "updatedAt": "2024-08-13T20:54:23.125Z",
          "id": "cB2FTZEBYaDeA-NhPHqW"
        },
        {
          "timestamp": "2024-08-13T20:54:23.125Z",
          "createdAt": "2024-08-13T20:54:23.125Z",
          "field": "agent.name",
          "allowed": true,
          "anonymized": true,
          "updatedAt": "2024-08-13T20:54:23.125Z",
          "id": "cR2FTZEBYaDeA-NhPHqW"
        }
      ],
      "deleted": [],
      "skipped": []
    },
    "summary": {
      "failed": 0,
      "succeeded": 2,
      "skipped": 0,
      "total": 2
    }
  }
}
```

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 35]
```diff
@@ -0,0 +1,233 @@
...
+--------------------------------------------------
+
+*Example 2: Partial failure*
+
+The following request removes anonymization field that have the ID sent in the payload and updates `allowed` value for the anonymization field with ID "lh12SZEBYaDeA-NhmkwG" and `anonymized` for ID "lR12SZEBYaDeA-NhmkwG":
```
Comment (natasha-moore-elastic) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:145]:
```suggestion
The following request:
* deletes the anonymization field with the ID value of `cR2FTZEBYaDeA-NhPHqW`
* updates the `allowed` value for the anonymization field with the ID of `lh12SZEBYaDeA-NhmkwG`
* updates the `anonymized` value for the anonymization field with the ID of `lR12SZEBYaDeA-NhmkwG`
```

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 36]
```diff
@@ -0,0 +1,233 @@
...
+      {
+         "id":"lh12SZEBYaDeA-NhmkwG",
+         "allowed":false
+      },
+      {
+         "id":"lR12SZEBYaDeA-NhmkwG",
+         "anonymized":true
+      }
+   ]
+}
```
Comment (natasha-moore-elastic) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:165]:
```suggestion
{
  "delete": {
    "ids": [
      "cR2FTZEBYaDeA-NhPHqW"
    ]
  },
  "update": [
    {
      "id": "lh12SZEBYaDeA-NhmkwG",
      "allowed": false
    },
    {
      "id": "lR12SZEBYaDeA-NhmkwG",
      "anonymized": true
    }
  ]
}
```

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 37]
```diff
@@ -0,0 +1,233 @@
...
+[discrete]
+===== Response code
+
+`500`::
+    Indicates partial bulk action failure.
+
+[discrete]
+===== Response payload
+
+If processing of any anonymization field fails, a partial error outputs the ID and/or name of the affected anonymization field and the corresponding error, as well as successfully processed anonymization fields (in the same format as a successful 200 request).
```
Comment (natasha-moore-elastic) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:177]:
```suggestion
If the processing of any anonymization fields fails, the response outputs a partial error, with the ID and/or name of the affected anonymization field and the corresponding error message. The response also includes successfully processed anonymization fields, in the same format as a successful `200` request.
```

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 38]
```diff
@@ -0,0 +1,233 @@
...
+            "skipped": [],
+        },
+        "summary": {
+            "failed": 1,
+            "succeeded": 1,
+            "skipped": 0,
+            "total": 2
+        }
+    }
+}
```
Comment (natasha-moore-elastic) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:232]:
```suggestion
{
  "message": "Bulk delete partially failed",
  "status_code": 500,
  "attributes": {
    "errors": [
      {
        "message": "Some error happened here",
        "status_code": 500,
        "anonymization_fields": [
          {
            "id": "cR2FTZEBYaDeA-NhPHqW",
            "field": "test"
          }
        ]
      }
    ],
    "results": {
      "updated": [
        {
          "timestamp": "2024-08-13T01:59:55.141Z",
          "createdAt": "2024-08-13T01:59:55.141Z",
          "field": "@timestamp",
          "allowed": false,
          "anonymized": false,
          "updatedAt": "2024-08-13T21:00:37.502Z",
          "namespace": "default"
        },
        {
          "timestamp": "2024-08-13T01:59:55.141Z",
          "createdAt": "2024-08-13T01:59:55.141Z",
          "field": "_id",
          "allowed": true,
          "anonymized": true,
          "updatedAt": "2024-08-13T21:00:37.502Z",
          "namespace": "default"
        }
      ],
      "created": [],
      "deleted": [],
      "skipped": []
    },
    "summary": {
      "failed": 1,
      "succeeded": 1,
      "skipped": 0,
      "total": 2
    }
  }
}
```

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 39]
```diff
@@ -0,0 +1,233 @@
...
+
+
+[discrete]
+[[bulk-update-anonymization-fields-schema]]
+===== `BulkUpdateAction` schema
+
+[width="100%",options="header"]
+|==============================================
+|Name |Type |Description |Required
+
```
Comment (natasha-moore-elastic) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:47]: Should there be an `id` field defined here? In the example 2 request, the `BulkUpdateAction` objects have `id` fields:

```
  "update": [
    {
      "id": "lh12SZEBYaDeA-NhmkwG",
      "allowed": false
    },
    {
      "id": "lR12SZEBYaDeA-NhmkwG",
      "anonymized": true
    }
  ]
```

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 40]
```diff
@@ -0,0 +1,233 @@
...
+    Indicates partial bulk action failure.
+
+[discrete]
+===== Response payload
+
+If processing of any anonymization field fails, a partial error outputs the ID and/or name of the affected anonymization field and the corresponding error, as well as successfully processed anonymization fields (in the same format as a successful 200 request).
+
+[discrete]
+==== Example payload
+
```
Comment (natasha-moore-elastic) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:181]:
```suggestion
```

File: docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc [hunk 41]
```diff
@@ -0,0 +1,241 @@
...
+===== Request body
+
+A JSON object with the following properties:
+
+[width="100%",options="header"]
+|==============================================
+|Name |Type |Description |Required
+| `create` | <<bulk-create-object-schema, BulkCreateAction[]>> |  Array of prompts objects to create. | No
+| `delete` | String[] |  Array of prompts IDs to which a bulk action will be applied. | No
+| `update` | <<bulk-update-object-schema, BulkUpdateAction[]>> |  Array of prompts objects with the fields to update. | No
```
[the remark this answers is not in the document]
Comment (YulNaumenko) [reply docs/AI-for-security/api/bulk-actions-prompts-api.asciidoc]: Yes, user can send the empty request body. Nothing will happen, they will get 200 OK, and 0 changes in the updated, deleted, created response props. Total changes will be 0.

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 42]
```diff
@@ -0,0 +1,232 @@
...
+      "total": 2
+    }
+  }
+}
+--------------------------------------------------
+
+*Example 2: Partial failure*
+
+The following request:
+* deletes the anonymization field with the ID value of `cR2FTZEBYaDeA-NhPHqW`
```
Comment (benironside) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:145]:
```suggestion

* deletes the anonymization field with the ID value of `cR2FTZEBYaDeA-NhPHqW`
```

**Review by benironside — APPROVED** [review]

File: docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc [hunk 43]
```diff
@@ -0,0 +1,233 @@
...
+--------------------------------------------------
+
+*Example 2: Partial failure*
+
+The following request:
+
+* deletes the anonymization field with the ID value of `cR2FTZEBYaDeA-NhPHqW`
+* updates the `allowed` value for the anonymization field with the ID of `lh12SZEBYaDeA-NhmkwG`
+* updates the `anonymized` value for the anonymization field with the ID of `lR12SZEBYaDeA-NhmkwG`
+[source,console]
```
Comment (natasha-moore-elastic) [inline docs/AI-for-security/api/bulk-actions-anonymization-fields-api.asciidoc:149]:
```suggestion
* updates the `anonymized` value for the anonymization field with the ID of `lR12SZEBYaDeA-NhmkwG`

[source,console]
```

**Review by natasha-moore-elastic — APPROVED** [review]

**Pull request merged**